### DISCUSSION TITLE
How to render two micro frontends in the same page

### DISCUSSION THREAD

**Discussion by deepakchethan** [SAP/luigi#3304] [General]
In https://www.youtube.com/watch?v=EzQK0v7xBuE. It was mentioned that we can have multiple micro frontends in the same page. 

![Screenshot 2023-06-02 at 12 06 11 PM](https://github.com/SAP/luigi/assets/20895544/120a49f5-f2f9-4cb3-bfcd-6e01a56b4f3e)

But I don't see any examples of doing so. Can you please confirm if we can do it without having to rely on splitView

**Comment by ndricimrr** [comment]
Hi @deepakchethan,
It is possible to do this without having to rely on splitview. 
With Luigi this can be done using [webcomponent](https://docs.luigi-project.io/docs/web-component) based Microfrontends through the [compound](https://docs.luigi-project.io/docs/navigation-parameters-reference/?section=compound) configuration option. 
Take a look at this FAQ question on our docu to start with:
[_Can I place a micro frontend within another micro frontend?_](https://docs.luigi-project.io/docs/faq/?section=can-i-place-a-micro-frontend-within-another-micro-frontend)

We provide an example of this feature in our [e2e test application](https://github.com/SAP/luigi/tree/main/test/e2e-test-application), more specifically at [this part](https://github.com/SAP/luigi/blob/de83a17bc0fd424d9867ceab936a352b944c473f/test/e2e-test-application/src/luigi-config/extended/projectDetailNav.js#L11) of the Luigi config.

**Reply by cufos** [reply]
Why when i tried to do the same things that you have specified in the example doesn;t work ? The browser shows me always the same error, Failed to resolve module specifier '/sampleapp.html#/webwc'. The base URL is about:blank because import() is called from a CORS-cross-origin scrip. Even if i have the flag validWebcomponentUrls:['.*?'] in the config. How may i solve the problem. Because i did't find any solution.

**Reply by ndricimrr** [reply]
Hi @cufos ,
This might happen if you are importing Luigi library from an external source (like a CDN). Could you let us know how you are  importing the Luigi library in your case.

**Reply by cufos** [reply]
ah ok so that's the problem, yeah because we're importing the Luigi library from a cdn, thanks.

**Reply by cufos** [reply]
I have only one question, would i import a microfrontend inside of a web component ? Or there is no way to do that. and sorry to trouble you. thanks.

**Reply by hardl** [reply]
with luigi compound, you don't import particular micro frontends in the "main" web component but instead define slots (see e.g. https://developer.mozilla.org/en-US/docs/Web/HTML/Element/slot). These slots will then be filled by luigi with the corresponding compound children from your luigi config.

**Comment by alexandra-simeonova** [comment]
Thank you for your question! I will reopen the discussion so it can serve as future reference in case anyone else has the same question and looks through our Discussions page.
